### PR TITLE
Fix omrfileTest failures when running as root

### PR DESCRIPTION
**Pull request by EricYangIBM**
PortFileTest2.file_test27 and PortFileTest2.file_test29 fail when runnning
as root. Add logic to change test behaviour if tests are being run as root.

Signed-off-by: Eric Yang <eric.yang@ibm.com>

### CODE REVIEW

**Comment by EricYangIBM** [comment]
The only changes are the checks for euid and the change from `rc` -> `fd2` in the error messages for (erroneously) successfully opening read only file for writing and vice versa, which is an error as the error message specifically has `fd` in it.

For reference, the failures:
```
6: /root/docker_hostdir/temp/omr/fvtest/porttest/omrfileTest.cpp line 3498: omrfile_test27 opened read-only file for writing fd=0 expected -1
6: 
6:              LastErrorNumber: -108
6:              LastErrorMessage: No such file or directory
6: 
6: /root/docker_hostdir/temp/omr/fvtest/porttest/testHelpers.cpp:109: Failure
6: Value of: 0 == numberFailedTestsInComponent
6:   Actual: false
6: Expected: true
6: Test failed!
6: [  FAILED  ] PortFileTest2.file_test27 (0 ms)
6: /root/docker_hostdir/temp/omr/fvtest/porttest/omrfileTest.cpp line 3626: omrfile_test29 omrfile_chown() to root succeeded, should have failed
6: 
6:              LastErrorNumber: -108
6:              LastErrorMessage: No such file or directory
6: 
6: /root/docker_hostdir/temp/omr/fvtest/porttest/testHelpers.cpp:109: Failure
6: Value of: 0 == numberFailedTestsInComponent
6:   Actual: false
6: Expected: true
6: Test failed!
6: [  FAILED  ] PortFileTest2.file_test29 (0 ms)
```

**Review by babsingh — CHANGES_REQUESTED** [review]

File: fvtest/porttest/omrfileTest.cpp [hunk 1]
```diff
@@ -3461,71 +3461,78 @@ TEST_F(PortFileTest2, file_test27)
...
-	fd1 = FILE_OPEN_FUNCTION(OMRPORTLIB, localFilename, EsOpenCreate | EsOpenWrite, 0666);
-	if (-1 == fd1) {
-		outputErrorMessage(PORTTEST_ERROR_ARGS, "%s() failed\n", FILE_OPEN_FUNCTION_NAME);
-	} else {
-		(void)FILE_CLOSE_FUNCTION(OMRPORTLIB, fd1);
-		for (m = 0; m < nModes; ++m) {
+#if !defined(OMR_OS_WINDOWS)
+	/* Do not run following test as a root user (euid == 0) as root can bypass file permissions. */
+	if (0 != omrsysinfo_get_euid()) {
+#endif /* !defined(OMR_OS_WINDOWS) */
```
Comment (babsingh) [inline fvtest/porttest/omrfileTest.cpp:3467]: the `ifdef` at the closing brace can be avoided.
```suggestion
#if !defined(OMR_OS_WINDOWS)
	/* Do not run following test as a root user (euid == 0) as root can bypass file permissions. */
	if (0 != omrsysinfo_get_euid())
#endif /* !defined(OMR_OS_WINDOWS) */
	{
```

**Review by babsingh — CHANGES_REQUESTED** [review]

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
jenkins build win

**Comment by babsingh** [comment]
Restarting the zOS job since it timed out in finding a node.

**Review by babsingh — APPROVED** [review]

**Comment by babsingh** [comment]
jenkins build zos

**Comment by jdmpapin** [comment]
Jenkins build zos

**Review by jdmpapin — APPROVED** [review]

**Pull request merged**